### ISSUE TITLE
om staged-config does not dump all configuration for new-relic-service-broker

### ISSUE THREAD

**Issue by jli1972** [pivotal-cf/om#448]
om version - 4.2.1
opsman version - 2.4.20

issues:
1) missing the insightsInsertKey -- it's null
2) does not include <oldPlan> and other attributes below:

```diff
--- newrelic-broker-current.yml.sorted	2019-11-04 16:54:12.502700200 +0000
+++ interpolated-files/environments/sea-ops/config/p-new-relic/p-new-relic.yml	2019-11-04 16:53:24.000000000 +0000
@@ -21,7 +21,15 @@ product-properties:
   .properties.nrplans:
     value:
     - insightsInsertKey: null
+      licenseKey: 12341234******
+      oldPlan: true
       orgs: null
+      planDescription: New Relic APM supports real-time instrumentation and monitoring
+        of application performance via its embedded agent that runs with apps. New
+        Relic agent code embedded or bundled with the app instruments the app code
+        and publishes performance monitoring metrics and other details back to New
+        Relic APM Dashboard.
+      planName: NewRelic-APM-PROD
       planOldGuid: 8910b747-1c03-3ef8-9461-cadd523ac1c8
       rpmAccountId: null
   .properties.org:
```

**Comment by kcboyle** [comment]
looking at the metadata for this tile:
```
    - description: WARNING -- DO NOT CHANGE THIS PROPERTY -- this property is automatically set by the tile if the plan was migrated from tile 1.12.12 or older
      label: pre-1.12.12 plan?
      reference: oldPlan
    - description: Associated license key for the Plan
      label: New Relic License Key
      reference: licenseKey
    - description: Name of Plan
      label: Plan Name
      reference: planName
```

`staged-config` will never include the planDescription when it is returned.
`licenseKey`, `oldPlan`, and `planName` do not appear to be a key you can set in the UI. If the fields cannot be set (in new relic's case, it looks like the configurable fields are explicitly set to `configurable: true` in the metadata. I would assume anything without this param set is `configurable: false`) it is _not_ included in the `staged-config`. This is intentional to allow the same experience when interacting with the tile in Ops Manager as the tile using `om`.

That said, interacting with the tile in the UI, the team is struggling to understand how the new relic tile works (it is third party, and not setup like some of the usual tile teams' tiles). How does Plan Name get populated when you add the tile? If there are errors while saving the inputs, `staged-config` doesn't know about the properties. Can you confirm that you got no errors when saving off the service broker properties in the UI? Can you also verify that you ran `om staged-config --no-redact`? (`--no-redact` will allow secrets and keys to be displayed)
<img width="941" alt="Screen Shot 2019-11-12 at 10 02 03 AM" src="https://user-images.githubusercontent.com/1361223/68692959-a355c680-0533-11ea-8def-3bb7876271c6.png">

**Comment by iplay88keys** [comment]
Looking into this again on a 2.4 Ops Manager, we discovered a few more things:
1. The screenshot @kcboyle attached was a Ops Manager UI fix sometime after 2.4. 
2. When you configure a plan in the New Relic Service Broker, it is no longer configurable in the API (though the UI makes it look like it is configurable until you deploy the broker). If you hit the `api/v0/staged/products/newrelic-broker-<GUID>/properties` endpoint, you get something like:

```
"planName": {
  "type": "string",
  "configurable": false,
  "credential": false,
  "value": "plan-name",
  "optional": false
}
```

The reason that staged-config doesn't return anything for those properties is that it specifically does not return non-configurable properties because Ops Manager won't respect them when they are set using configure-product.

A workaround is to manually add the missing properties to the  `.properties.nrplans:` section of the output from staged-config:
```
product-properties:
  .properties.nrplans:
    value:
    - insightsInsertKey: some-insert-key
      orgs: some-org
      planOldGuid: some-old-guid
      rpmAccountId: some-account-id
      planName: some-plan-name
      planDescription: some-description
      licenseKey: some-license-key
```

Note that this will not work if the New Relic tile has already been deployed. It will also overwrite any existing collections if you haven't deployed it yet.

This might be a bug with the New Relic tile and how their collection properties for the plans doesn't have `configurable: true` on some of them, but the om cli is handling this case appropriately. We recommend reaching out to the Pivotal Tile Support team in order to get it fixed in the New Relic Tile.
